core/src/main/java/io/aiven/klaw/service/EnvsClustersTenantsControllerService.java

```diff
@@ -1245,39 +1248,52 @@ public KwReport getPublicKey() {
     return kwPublicKey;
   }
 
-  public EnvUpdatedStatus getUpdateEnvStatus(String envId) throws KlawException {
+  public EnvUpdatedStatus getUpdateEnvStatus(String envId) throws KlawBadRequestException {
     EnvUpdatedStatus envUpdatedStatus = new EnvUpdatedStatus();
     int tenantId = commonUtilsService.getTenantId(getUserName());
-    Env env = manageDatabase.getHandleDbRequests().getEnvDetails(envId, tenantId);
+    List<Env> allEnvs = manageDatabase.getAllEnvList(tenantId);
+    Optional<Env> env =
+        allEnvs.stream()
+            .filter(e -> e.getId().equals(envId) && e.getTenantId().equals(tenantId))
```

Comment:
Does it help to add condition e.getEnvExists ?

Reply:
I believe that looking at how the envs are loaded into the cache only those where getEnvExists==true will be loaded. So adding that check will always return true.